probe-rs/src/bin/probe-rs/main.rs

```diff
@@ -92,22 +93,22 @@ pub(crate) struct CoreOptions {
 }
 
 /// A helper function to deserialize a default [`Format`] from a string.
-fn format_from_str<'de, D: Deserializer<'de>>(deserializer: D) -> Result<Format, D::Error> {
+fn format_from_str<'de, D: Deserializer<'de>>(deserializer: D) -> Result<Option<Format>, D::Error> {
     match Value::deserialize(deserializer)? {
         Value::String(s) => match Format::from_str(s.as_str()) {
-            Ok(format) => Ok(format),
+            Ok(format) => Ok(Some(format)),
             Err(e) => Err(D::Error::custom(e)),
         },
-        _ => Err(D::Error::custom("invalid format")),
+        _ => Ok(None),
     }
 }
 
 #[derive(clap::Parser, Clone, Deserialize, Debug, Default)]
 #[serde(default)]
 pub struct FormatOptions {
-    #[clap(value_enum, ignore_case = true, default_value = "elf", long)]
+    #[clap(value_enum, ignore_case = true, long)]
     #[serde(deserialize_with = "format_from_str")]
-    format: Format,
+    format: Option<Format>,
```

Comment:
Please document the format which is used if nothing is specified.

Reply:
I've documented this on the `into_format` method.

Reply:
The motivation for documenting it here is to show it in the CLI help, so documenting `into_format` doesn't help.

Reply:
Ah yeah, as you can tell I don't use clap much :D. Fixed.